tests/rbd_mirror/test_mirror_move_primary_trash_restore.py

```diff
@@ -0,0 +1,103 @@
+from tests.rbd.rbd_utils import Rbd
+from tests.rbd_mirror import rbd_mirror_utils as rbdmirror
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(**kw):
+    """Verification of trash restore on an image with snap and clone in primary cluster
+    to check that mirror relationship is intact post restore.
+
+    Args:
+        **kw:
+
+    Returns:
+        0 - if test case pass
+        1 - it test case fails
+
+    Test case covered -
+    CEPH-11417 - Pool Mirror - verify mirroring is intact after restore primary
+    image having snap and clone configured from trash
+
+    Pre-requisites :
+    1. At least two clusters must be up and running with enough number of OSDs to create pools
+    2. We need atleast one client node with ceph-common package,
+       conf and keyring files
+
+    Test Case Flow:
+    1. Create an image in primary and perform IOs.
+    2. enable pool mirroring on the image.
+    3. verify for the pool mirror mode enabled on secondary.
+    4. Create anapshot and clone from the snapshot on primary.
+    5. Move the primary image to the trash.
+    6. Restore the primary image from the trash.
+    7. Make sure that image is restored and the mirror relationship is intact.
+    """
+    try:
+        log.info("Starting RBD mirroring test case - CEPH-11417")
+        config = kw.get("config")
+
+        mirror1, mirror2 = [
+            rbdmirror.RbdMirror(cluster, config)
+            for cluster in kw.get("ceph_cluster_dict").values()
+        ]
+
+        rbd1, rbd2 = [
+            Rbd(**kw, req_cname=cluster_name)
+            for cluster_name in kw.get("ceph_cluster_dict").keys()
+        ]
+
+        poolname = mirror1.random_string() + "_11417pool"
```

Comment:
after random string `_11417` will be redundant.

Reply:
done.